ARCHITECTURE.md

```diff
@@ -0,0 +1,3319 @@
+            .collect()
+    }
+}
+
+/// The assembled context ready for prompt generation
+#[derive(Debug, Clone)]
+pub struct AssembledContext {
+    pub event_description: String,
+    pub relevant_facts: Vec<String>,
+    pub world_context: WorldContext,
+    pub character_context: Vec<CharacterContext>,
+    pub activated_tags: Vec<(Tag, f32)>,
+}
+
+#[derive(Debug, Clone)]
+pub struct WorldContext {
+    pub time_of_day: String,
+    pub day: u32,
+    pub season: Season,
+    pub weather: Weather,
+    pub is_night: bool,
+}
+
+#[derive(Debug, Clone)]
+pub struct CharacterContext {
+    pub name: String,
+    pub title: Option<String>,
+    pub current_hp_percent: u32,
+    pub active_statuses: Vec<String>,
+    pub personality: Vec<String>,
+}
+```
+
+**Spreading Activation Visualization:**
+
+```mermaid
+flowchart TD
+    subgraph "Step 1: Trigger"
+        E[GameEvent: CombatAbilityUsed]
+        E --> T1[Villain_A: 1.0]
+        E --> T2[Priestess_B: 1.0]
+        E --> T3[Magic: 1.0]
+    end
+
+    subgraph "Step 2: First Spread"
+        T1 -->|0.9 × 0.5| T4[Orphanage: 0.45]
+        T2 -->|0.9 × 0.5| T4
+        T1 -->|0.7 × 0.5| T3
+        T2 -->|0.8 × 0.5| T3
+    end
+
+    subgraph "Step 3: Accumulate"
+        T4 --> ACC[Orphanage: 0.9]
+        T3 --> ACC2[Magic: 1.75]
+    end
+
+    subgraph "Step 4: Collect Facts"
+        ACC --> F1[Fact: Grew up together]
+        ACC2 --> F2[Fact: Both have magical abilities]
+    end
+```
+
+#### 2.2.3 Module: `llm_interface`
+
+**Purpose:** Communicate with LLM (Ollama) and manage tool definitions. AI always uses typed JSON output.
+
+> **Note:** Consider using the `schemars` crate to auto-generate JSON schemas from Rust types.
+
+**AI Response Types:**
+
+AI always responds with structured, typed JSON. This enables:
+- Generating dialog variants for player selection
+- Typed emotional tone and suggested actions
+- Tool calls for fact recording
+
+```rust
+// src/llm_interface/response_types.rs
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+/// AI response for narrative generation
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
+pub struct NarrativeResponse {
+    /// The narrative text to display
+    pub narrative: String,
+
+    /// Emotional tone of this response
+    pub tone: EmotionalTone,
+
+    /// Suggested player actions/responses
+    pub suggested_actions: Vec<SuggestedAction>,
+
+    /// Tool calls to execute
+    pub tool_calls: Vec<ToolCall>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
+pub enum EmotionalTone {
+    Neutral,
+    Friendly,
+    Hostile,
+    Fearful,
+    Mysterious,
+    Sad,
+    Joyful,
+}
+
+/// Dialog variant for player selection
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
+pub struct DialogVariant {
+    pub id: u8,
+    pub text: String,
+    pub tone: EmotionalTone,
+    /// Potential consequences (for UI hints)
+    pub consequence_hint: Option<String>,
+}
```

Comment:
that also generate by ai, maybe `id` is useless because it comes from list always